specifications/api/swagger.yaml

```diff
@@ -2532,17 +2560,35 @@ components:
               type: string
               enum: ["native"]
             script: *ScriptValue
+            reference: *referenceInput
```

Comment:
When we have spending script tx, tx has script (not hash of it, but full form) in witness. Ledger needs a script to be included in witness set because otherwise it cannot realize its rules. hash is not enough!. 

In case of reference scripts they situation is different. Definition of script is in TxOut, but it ACTS as a witness in the first case. And this is realized in separate earlier tx. The consumption of this tx can be realized in later tx, not necessairly one, but could be more than one. In that case tx uses the reference to the script of the first tx. In order to be consistent with "spending script" mode I made the script part of witness during DEFINITION of it. I know it is part of TxOut but it acts as witness really! I think when we support fully reference scripts (so on the levelof constructing tx with them), we will need to extend out TxOut and in that case script will need to be part of TxOut. This extension of TxOut is big change to our codebase. Still the script included in TxOut will act as witness. 

What I am saying is that including scripts and also scripts to be referenced later in WitnessCount is reasonable to me as it reflects the nature and function of this script. Also WitnessCount could be inspected in any tx, not signed, partially-signed and fully-signed. It just underline witness detected in a given cbor. So in reference tx it detects script to be used later in other txs! And this is fine IMHO